### DISCUSSION TITLE
\colorbox does not work inside of \gdef

### DISCUSSION THREAD

**Discussion by rudolf-adamkovic** [KaTeX/KaTeX#2562] [Q&A] [answered]
I am struggling to define a `\highlight` macro.

`\colorbox` works:
```
\colorbox{#FFE270}{Hi.}
```

`\gdef` works:
```
\gdef{\abs}#1{\lvert #1 \rvert}
\abs{5}
```

`\gdef` with `\colorbox` does not work:
```
\gdef{\highlight}#1{\colorbox{#FFE270}{#1}}
\highlight{Hi.}
```

All three macros (KaTeX in iA Writer) render as follows:

<img width="443" alt="Screen Shot 2020-10-03 at 00 23 35" src="https://user-images.githubusercontent.com/1158274/94974366-afad8e00-050e-11eb-971b-08be5206e536.png”>

Am I doing something wrong? Is this supposed to work?

**Comment by edemaine** [comment, marked as answer]
In TeX, a literal `#` inside a macro needs to be escaped as `##`. Try this:

```latex
\gdef{\highlight}#1{\colorbox{##FFE270}{#1}}
\highlight{Hi.}
```

**Reply by rudolf-adamkovic** [reply]
That worked. Thank you so much!
